gpflow/kernels/linears.py

```diff
@@ -13,24 +13,23 @@ class Linear(Kernel):
     where σ²  is the variance parameter.
     """
 
-    def __init__(self, variance=1.0, active_dims=None, ard=None):
+    def __init__(self, variance=1.0, active_dims=None):
         """
-        - input_dim is the dimension of the input to the kernel
-        - variance is the (initial) value for the variance parameter(s)
-          if ard=True, there is one variance per input
-        - active_dims is a list of length input_dim which controls
-          which columns of X are used.
+        :param variance: the (initial) value for the variance parameter(s),
+            to induce ARD behaviour this must be initialised as an array the same
+            length as the the number of active dimensions e.g. [1., 1., 1.]
+        :param active_dims: a slice or list specifying which columns of X are used
         """
         super().__init__(active_dims)
-
-        # variance, self.ard = self._validate_ard_shape("variance", variance, ard)
-        self.ard = ard
         self.variance = Parameter(variance, transform=positive())
+        self.ard = self._is_ard(variance)
```

Comment:
I think it's better as a property (as you had before), e.g. if someone decides to replace the `variance` attribute with a different Parameter object. Same applies to the other kernels as well of course!

Reply:
Good point - I have reintroduced `ard` as a parameter and improved the implementation by creating an `ARDKernel` base class to enable better code re-use.